src/effects/builtin/tonegen/toneviewmodel.cpp

```diff
@@ -39,53 +42,105 @@ QList<QString> ToneViewModel::waveforms() const
     };
 }
 
-double ToneViewModel::amplitude() const
+QList<QString> ToneViewModel::interpolationTypes() const
+{
+    return {
+        tr("Linear"), tr("Logarithmic")
+    };
+}
+
+double ToneViewModel::amplitudeStart() const
 {
     const ToneEffect* const te = effect();
     if (!te) {
         return 0.0;
     }
-    return te->amplitude();
+    return te->amplitudeStart();
 }
 
-void ToneViewModel::prop_setAmplitude(double newAmplitude)
+void ToneViewModel::prop_setAmplitudeStart(double newAmplitude)
 {
     if (!m_inited) {
         return;
     }
-
     const auto wasAllowed = isApplyAllowed();
     ToneEffect* const te = effect();
     IF_ASSERT_FAILED(te) {
         return;
     }
-    te->setAmplitude(newAmplitude);
+    te->setAmplitudeStart(newAmplitude);
     if (wasAllowed != isApplyAllowed()) {
         emit isApplyAllowedChanged();
     }
 }
 
-double ToneViewModel::frequency() const
+double ToneViewModel::amplitudeEnd() const
 {
     const ToneEffect* const te = effect();
     if (!te) {
         return 0.0;
     }
-    return te->frequency();
+    return te->amplitudeEnd();
 }
 
-void ToneViewModel::prop_setFrequency(double newFrequency)
+void ToneViewModel::prop_setAmplitudeEnd(double newAmplitude)
 {
     if (!m_inited) {
         return;
     }
+    const auto wasAllowed = isApplyAllowed();
+    ToneEffect* const te = effect();
+    IF_ASSERT_FAILED(te) {
+        return;
+    }
+    te->setAmplitudeEnd(newAmplitude);
+    if (wasAllowed != isApplyAllowed()) {
+        emit isApplyAllowedChanged();
+    }
+}
+
+double ToneViewModel::frequencyStart() const
+{
+    const ToneEffect* const te = effect();
+    if (!te) {
+        return 0.0;
+    }
+    return te->frequencyStart();
+}
+
+void ToneViewModel::prop_setFrequencyStart(double newFrequency)
+{
+    const auto wasAllowed = isApplyAllowed();
```

Comment:
Shouldn't you add
```cpp
    if (!m_inited) {
        return;
    }
```
here too? (Explanation in [this comment](https://github.com/audacity/audacity/pull/7927#issue-2745281322))